src/util.cpp

```diff
@@ -209,10 +209,10 @@ void EnsureDirExists(const std::string &filepath) {
 std::string AbsolutePath(const std::string &filepath) {
   // clang-format off
 
-  #ifdef FLATBUFFERS_NO_ABSOLUTE_PATH_RESOLUTION
+  #if defined(FLATBUFFERS_NO_ABSOLUTE_PATH_RESOLUTION) || defined(__CYGWIN__) || defined(__MINGW32__) || defined(__MINGW64__)
```

Comment:
Why are we disabling getting the absolute path for those compilers?

Reply:
Both Cygwin and MinGW don't have `realpath()` function. Compiler generates the corresponding error:
```
flatbuffers/src/util.cpp:219:29: error: ‘realpath’ was not declared in this scope
  219 |       char *abs_path_temp = realpath(filepath.c_str(), nullptr);
      |                             ^~~~~~~~
```

Reply:
But `GetFullPathNameA` still works?

The logic for the preprocessor conditionals is just wrong. If MINGW23 is defined, it just `return filepath` which is not what we want.

Reply:
Yes, `GetFullPathNameA` should work, but we need to enable Win32API includes for MinGW & Cygwin. Currently it enabled only for _WIN32
```
#ifdef _WIN32
#  ifndef WIN32_LEAN_AND_MEAN
#    define WIN32_LEAN_AND_MEAN
#  endif
#  ifndef NOMINMAX
#    define NOMINMAX
#  endif
#  ifdef _MSC_VER
#    include <crtdbg.h>
#  endif
#  include <windows.h>  // Must be included before <direct.h>
#  include <direct.h>
#  include <winbase.h>
#  undef interface  // This is also important because of reasons
#endif
```

Reply:
Let my try it in my fork